### PR TITLE
fix multicategorical stype inference and add test case

### PR DESCRIPTION
**Pull request by yiweny**

### CODE REVIEW

File: torch_frame/utils/infer_stype.py [hunk 1]
```diff
@@ -138,18 +139,25 @@ def infer_series_stype(ser: Series) -> stype | None:
...
+                                    lambda row: MultiCategoricalTensorMapper.
+                                    split_by_sep(row, sep)).explode()))
+                    except Exception as e:
+                        logging.warn(
+                            "Mapping series into multicategorical stype "
+                            f"with separator {sep} raised an exception {e}")
+                        continue
+
+            if len(min_count_list) > 0 and max(
+                    min_count_list) > cat_min_count_thresh:
```
Comment (rusty1s) [inline torch_frame/utils/infer_stype.py:160]: don't you need a unique here?

**Review by weihua916 — APPROVED** [review]

File: torch_frame/utils/infer_stype.py [hunk 2]
```diff
@@ -138,18 +139,25 @@ def infer_series_stype(ser: Series) -> stype | None:
...
+                        min_count_list.append(
+                            _min_count(
+                                ser.apply(
+                                    lambda row: MultiCategoricalTensorMapper.
+                                    split_by_sep(row, sep)).explode()))
+                    except Exception as e:
+                        logging.warn(
+                            "Mapping series into multicategorical stype "
+                            f"with separator {sep} raised an exception {e}")
+                        continue
```
Comment (weihua916) [inline torch_frame/utils/infer_stype.py:157]: nit: I would put `min_count_list` inside else:
then do `max(min_count_list)` in else statement. in the if-statement, I would just put the value of `_min_count(ser.explode())`.

Then judge if the value is larger than `cat_min_count_thresh`

Comment (yiweny) [reply torch_frame/utils/infer_stype.py:157]: got it. Fixed

File: torch_frame/utils/infer_stype.py (hunk 1 above)
Comment (yiweny) [reply torch_frame/utils/infer_stype.py]: What do you mean by need a unique?

**Review by weihua916 — APPROVED** [review]

**Pull request merged**